public/components/metrics/sidebar/__tests__/sidebar.test.tsx

```diff
@@ -37,16 +41,59 @@ describe('Side Bar Component', () => {
       })
     );
 
-    const wrapper = mount(
-      <Provider store={store}>
-        <Sidebar />
-      </Provider>
-    );
+    let wrapper: ReactWrapper;
+    act(() => {
+      wrapper = mount(
+        <Provider store={store}>
+          <Sidebar />
+        </Provider>
+      );
+    });
 
-    wrapper.update();
+    expect(wrapper.text()).toContain('Available Metrics 0 of 0');
+
+    await act(async () => {
+      console.log('calling resolver');
+      objectListResolver({ observabilityObjectList: [sampleSavedMetric] });
+    });
 
     await waitFor(() => {
-      expect(wrapper).toMatchSnapshot();
+      wrapper.update();
+      expect(wrapper.text()).toContain('Available Metrics 1 of 1');
+    });
+
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it.skip('renders Side Bar Component for Saved Metric', async () => {
+    let objectListResolver;
+    SavedObjectsActions.getBulk = () =>
+      new Promise((_resolve) => {
+        objectListResolver = _resolve;
+      });
+
+    httpClientMock.get = jest.fn();
+
+    coreRefs.pplService = new PPLService(httpClientMock);
+    coreRefs.pplService.fetch = jest.fn(() => {
+      console.log('mock pplService fetch');
+      return new Promise((resolve) => {
+        resolve({
+          data: { DATA_SOURCES: ['datasource1', 'datasource2'] },
+          then: () => Promise.resolve(),
+        });
+      });
+    });
+
+    const el = document.createElement('div');
+    act(() => {
+      ReactDOM.render(<Sidebar />, el);
+    });
+    expect(el.innerHTML).toContain('Available Metrics 0 of 0');
+    await act(async () => {
+      objectListResolver({ observabilityObjectList: [sampleSavedMetric] });
     });
+    expect(el.innerHTML).toContain('Available Metrics 1 of 1');
```

Comment:
In a lot of the flaky tests I've been working with lately, I've seen a lot of these types of string assertions. Fine, but I wonder if there's a better/more stable way to check this.

Reply:
This test is asserting that the output values (the literal rendered string "Available Metrics 1 of 1") are rendered after a side-effect update from redux store.

More stable than... ?
1. display it (the blank-ish state)
2. change something
3. assert the display changed